modules/dag-l0/src/main/scala/org/tessellation/dag/l0/domain/snapshot/programs/Download.scala

```diff
@@ -120,24 +119,29 @@ object Download {
       retryingOnSomeErrors(retryPolicy, isWorthRetrying, retry.noop[F, Throwable]) {
         val (lastSnapshot, lastContext) = result
 
-        fetchSnapshot(none, lastSnapshot.ordinal.next).flatMap { snapshot =>
-          lastSnapshot.toHashed[F].flatMap { hashed =>
-            Applicative[F].unlessA {
-              Validator.isNextSnapshot(hashed, snapshot.value)
-            }(InvalidChain.raiseError[F, Unit])
-          } >>
-            globalSnapshotContextFns
-              .createContext(lastContext, lastSnapshot, snapshot)
-              .handleErrorWith(_ => InvalidChain.raiseError[F, GlobalSnapshotContext])
-              .flatTap { _ =>
-                snapshotStorage.writePersisted(snapshot)
-              }
-              .map((snapshot, _))
+        hasherSelector.forOrdinal(lastSnapshot.ordinal.next) { implicit hasher =>
+          fetchSnapshot(none, lastSnapshot.ordinal.next).flatMap { snapshot =>
+            lastSnapshot.toHashed[F].flatMap { hashed =>
```

Comment:
This may be correct but it's confusing to know when we hit the boundary condition.
Why is `lastSnapshot` hashed with the hasher retrieved with `lastSnapshot.ordinal.next` instead of `lastSnapshot.ordinal` ?

Reply:
Let me take a closer look :+1: Thanks for spot

Reply:
The thing that occurred to me is when we hit the boundary we'll need 2 hashers, won't we?

Reply:
@sdavidp fixed